### PR TITLE
Update CI setup

### PR DESCRIPTION
**Pull request by sk1p**
## TODO

- [x] Rebase to master once #6 is merged
- [x] Add `LiberTEM-live-data` pipeline
- [x] Add secrets/tokens/IDs for releases
  - [x] PyPI
  - [x] Zenodo (deposition IDs and tokens)
  - [x] GitHub
- [x] Run example notebooks in CI (need simulator(s) etc.)

## Contributor Checklist:

* [N/A] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [N/A] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [N/A] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases

## Reviewer Checklist:

* [x] `/azp run LiberTEM.LiberTEM-live-data` passed

### CODE REVIEW

**Comment by sk1p** [comment]
> No pipelines are associated with this pull request.

Why not, though?

**Comment by sk1p** [comment]
> Why not, though?

Was still missing a private key, which I've now added. Some tokens for releasing are also still missing.

**Comment by sk1p** [comment]
This is now mostly waiting for the last TODO item, and for #6 to be merged. Adding notebook CI (and re-adding doctest support) can also be done in a separate step.

**Comment by sk1p** [comment]
`nbval` spewed this error, just for reference. Never got that before, let's keep this in mind:

```
2021-06-09T15:34:08.4791825Z 2021-06-09 15:33:18 ERROR Failure starting kernel ":nbval-parent-env"
2021-06-09T15:34:08.4792434Z Traceback (most recent call last):
2021-06-09T15:34:08.4793439Z   File "/azp/agent/_work/1/s/.tox/notebooks/lib/python3.8/site-packages/nbval/kernel.py", line 57, in start_new_kernel
2021-06-09T15:34:08.4794135Z     kc.wait_for_ready(timeout=startup_timeout)
2021-06-09T15:34:08.4795201Z   File "/azp/agent/_work/1/s/.tox/notebooks/lib/python3.8/site-packages/jupyter_client/blocking/client.py", line 111, in wait_for_ready
2021-06-09T15:34:08.4796236Z     raise RuntimeError('Kernel died before replying to kernel_info')
2021-06-09T15:34:08.4796832Z RuntimeError: Kernel died before replying to kernel_info
```

Retrying the job fixed the problem.

**Comment by sk1p** [comment]
A note on performance: right now, both ptychography40 and libertem are installed from git, so creating the tox environments is still quite slow; I expect this will speed up significantly once these two packages are released.

**Comment by sk1p** [comment]
Also, we may need to adjust some timeouts - many of the tests take exactly 10s in setup, which looks like it runs into some kind of timeout.

**Comment by sk1p** [comment]
This should now be ready to be merged, feel free to go ahead any time!

File: src/libertem_live/detectors/merlin/sim.py [hunk 1]
```diff
@@ -65,6 +66,17 @@ def __init__(self, host, port, name, *args, **kwargs):
         self._name = name
         self.listen_event = threading.Event()
 
+    def wait_for_listen(self, timeout=10):
+        """
+        To be called from the main thread
+        """
+        t0 = time.time()
+        while time.time() - t0 < timeout:
+            self.listen_event.wait(timeout=0.1)
```
Comment (sk1p) [inline src/libertem_live/detectors/merlin/sim.py:75]:
```suggestion
            if self.listen_event.wait(timeout=0.1):
                return
```

**Review by uellue — APPROVED** [review]
Thanks a lot, merging!

**Pull request merged**